core/templates/domain/topics_and_skills_dashboard/TopicsAndSkillsDashboardFilterObjectFactory.ts

```diff
@@ -14,42 +14,47 @@
 
 /**
  * @fileoverview Factory for creating filters of frontend topics
- * and skill dashboard
+ * and skill dashboard.
  */
 
 import { downgradeInjectable } from '@angular/upgrade/static';
 import { Injectable } from '@angular/core';
 
-import { ETopicPublishedOptions, ETopicSortOptions } from
+import {
+  ETopicPublishedOptions,
+  ETopicSortOptions,
+  TopicsAndSkillsDashboardPageConstants } from
   // eslint-disable-next-line max-len
   'pages/topics-and-skills-dashboard-page/topics-and-skills-dashboard-page.constants';
 
 export class TopicsAndSkillsDashboardFilter {
   category: string;
-  keyword: string;
+  keywords: Array<string>;
   sort: ETopicSortOptions;
   status: ETopicPublishedOptions;
 
   /**
    * @param {String} category - category to filter for.
-   * @param {String} keyword - keyword to filter for.
-   * @param {ETopicSortOptions} sort - Enum, allowed sort-by values.
-   * @param {ETopicPublishedOptions} status - Enum. allowed status values.
+   * @param {String[]} keywords - keywords to filter for.
+   * @param {ETopicSortOptions} sort - One of the values in ETopicSortOptions.
+   * @param {ETopicPublishedOptions} status - One of the values
+   *   in ETopicPublishedOptions.
    */
-  constructor(category, keyword, sort, status) {
+  constructor(category, keywords, sort, status) {
     this.category = category;
-    this.keyword = keyword;
+    this.keywords = keywords;
     this.sort = sort;
     this.status = status;
   }
   /**
    * Resets the filter object values
    */
   reset(): void {
-    this.category = '';
-    this.keyword = '';
-    this.sort = null;
-    this.status = null;
+    this.category =
+        TopicsAndSkillsDashboardPageConstants.TOPIC_FILTER_DEFAULT_VALUE;
```

Comment:
The naming seems off -- there is nothing in the name of this constant that relates to "category".

Reply:
Fixed.